src/components/TodoLists.jsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import NoneTodos from './NoneTodos';
+import ShowTodos from './ShowTodos';
+
+export default function TodoLists({ hasTodos, todos, onClickComplete }) {
+  return (
+    <>
+      {hasTodos ? (
+        <ShowTodos todos={todos} onClickComplete={onClickComplete} />
+      ) : (
+        <NoneTodos />
+      )}
+    </>
+  );
+}
```

Comment:
제가 의도했던 건 

```javascript
<li key={id}>
  <span>{text}</span>
  <button type="button" id={id} onClick={() => onClickComplete(id)}>
    완료
  </button>
</li>
```

이 부분을 빼는 것을 생각했었어요 ㅎㅎ 여기 코드는 3항 연산자도 좋지만 Guard clause로 작성하는 것도 좋을 것 같아요.

```suggestion
const isEmpty = (arr = []) => arr.length === 0;

export default function TodoLists({ hasTodos, todos, onClickComplete }) {
  if (isEmpty(todos)) {
    return <NoneTodos />;
  }

  return (
    <ShowTodos todos={todos} onClickComplete={onClickComplete} />
  );
}
```

See also
  - https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Reply:
헉 코드 정말 깔끔해졌네요!
저는 hasTodos라는 이름으로 todo의 유무도 컴포넌트 인자로 넘겨주면서 너무 투머치 한 느낌도 있었거든요
함수로 빼서 저렇게 사용하는 것도 좋네요!

그리고 처음에 짤 때 가드클래스를 사용했었는데 윤석님이 짜신 것처럼 바로 export default에 넣을 생각을 못했었는데.. 익숙치 않아서 그랬던 것 같습니다!

보내주신 코드에서 지향한 부분 앞으로 코드에도 잘 녹여서 진행해볼게요! 감사해요 :)